### ISSUE TITLE
Crash Bug: Changing Entity Editor Visuals

### ISSUE THREAD

**Issue by mazett96** [deepnight/ldtk#968]
Changing the editor visuals causes the editor to crash.

When Trying to change a rectangle to an elipse this is what is printed:

LDtk version: 1.4.0-64bits
Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius provided (-7) is negative. (IndexSizeError)

It continues to pop up, even when I try to do other changes, such as trying to add an icon form a tileset, but then it's more rare and doesnt happen every time.

<details><summary>Details</summary>
<p>

Stack:
```
LDtk version: 1.4.0-64bits
Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius provided (-7) is negative.
IndexSizeError
Error: Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius provided (-7) is negative.
    at misc_JsTools.createEntityPreview (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:203582:8)
    at ui_modal_panel_EditEntityDefs.updatePreview (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:269679:61)
    at ui_modal_panel_EditEntityDefs.selectEntity (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:268723:8)
    at HTMLLIElement.<anonymous> (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:269655:14)
    at HTMLLIElement.dispatch (C:\Users\mazet\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:43064)
    at v.handle (C:\Users\mazet\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:41048)
```
Processes:
```
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #13 tool.lt.EntityTool[--]
       |--- #49 display.EntityRender
       |--- #50 display.EntityRender
       |--- #51 display.EntityRender
       |--- #52 display.EntityRender
       |--- #53 display.EntityRender
       |--- #54 display.EntityRender
       |--- #55 display.EntityRender
       |--- #56 display.EntityRender
       |--- #57 display.EntityRender
       |--- #58 display.EntityRender
       |--- #59 display.EntityRender
       |--- #60 display.EntityRender
       |--- #61 display.EntityRender
       |--- #62 display.EntityRender
       |--- #63 display.EntityRender
       |--- #64 display.EntityRender
       |--- #65 display.EntityRender
       |--- #66 display.EntityRender
       |--- #67 display.EntityRender
       |--- #68 display.EntityRender
       |--- #69 display.EntityRender
       |--- #70 display.EntityRender
       |--- #71 display.EntityRender
       |--- #72 display.EntityRender
       |--- #73 display.EntityRender
       |--- #74 display.EntityRender
       |--- #75 display.EntityRender
       |--- #76 display.EntityRender
       |--- #77 display.EntityRender
       |--- #78 display.EntityRender
       |--- #79 display.EntityRender
       |--- #80 display.EntityRender
       |--- #81 display.EntityRender
       |--- #82 display.EntityRender
       |--- #83 display.EntityRender
       |--- #90 ui.modal.panel.EditEntityDefs
```

Log:
```
[CACHE]         -> identified as Png
[CACHE]         -> base64 20428
[CACHE]         -> pixels 188x352
[FILE]          -> Loaded 15321 bytes.
[FILE]          -> Decoded 188x352 pixels.
[FILE]          -> Old size: 188x352 -> 188x352
[WARNING]      Tileset remapping (image size changed)...
[FILE]          -> Reload result: Ok
[FILE]          -> pixelData: Ok
[EVENT]        TilesetImageLoaded
[FILE]         Reloading tileset: Internal_Icons path=null
[FILE]         Loading atlas image: LdtkIcons...
[FILE]         Loading embed atlas LdtkIcons: C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/embedAtlas/finalbossblues-icons_full_16.png ...
[FILE]          -> Loaded 195310 bytes.
[FILE]          -> Decoded 512x1024 pixels.
[FILE]          -> Old size: 512x1024 -> 512x1024
[WARNING]      Tileset remapping (image size changed)...
[FILE]          -> Reload result: Ok
[FILE]          -> pixelData: Ok
[EVENT]        TilesetImageLoaded
[EVENT]        ProjectSelected
[RENDER]       Rendered level UI
[RENDER]       Rendering all world...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bounds...
[FILE]         Watching file: C:\Users\mazet\Documents\GitHub\FailFailSucceed\FailFailSucceed\Assets\LDtkProject\WorldTiles.png
[RENDER]       Rendering world level data.Level#eb724e20-02f0-11ed-91a8-6b345142fc4a "Lobby"...
[RENDER]       Rendering world level fields data.Level#eb724e20-02f0-11ed-91a8-6b345142fc4a "Lobby"...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[USERACTION]   Opened panel #90 ui.modal.panel.EditEntityDefs
[FILE]         Loading HTML template editEntityDefs
[FILE]         Loading HTML template fieldDefsForm
[FILE]         Loading HTML template pivotEditor
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[RENDER]       Rendering world bg...
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius provided (-7) is negative. (IndexSizeError)
[ERROR]        Error: Failed to execute 'ellipse' on 'CanvasRenderingContext2D': The major-axis radius provided (-7) is negative.
    at misc_JsTools.createEntityPreview (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:203582:8)
    at ui_modal_panel_EditEntityDefs.updatePreview (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:269679:61)
    at ui_modal_panel_EditEntityDefs.selectEntity (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:268723:8)
    at HTMLLIElement.<anonymous> (file:///C:/Users/mazet/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:269655:14)
    at HTMLLIElement.dispatch (C:\Users\mazet\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:43064)
    at v.handle (C:\Users\mazet\AppData\Local\Programs\ldtk\resources\app.asar\assets\js\jquery.min.js:2:41048)

[GENERAL]      
#0 App
 |--- #3 page.Editor
       |--- #4 misc.FileWatcher
       |--- #5 display.WorldRender
       |--- #6 display.LevelRender
       |--- #7 display.Camera
       |--- #8 display.Rulers[--]
       |--- #9 tool.SelectionTool[--]
       |--- #10 tool.lt.DoNothing[--]
       |--- #11 WorldTool
       |--- #12 tool.PanView[--]
       |--- #13 tool.lt.EntityTool[--]
       |--- #49 display.EntityRender
       |--- #50 display.EntityRender
       |--- #51 display.EntityRender
       |--- #52 display.EntityRender
       |--- #53 display.EntityRender
       |--- #54 display.EntityRender
       |--- #55 display.EntityRender
       |--- #56 display.EntityRender
       |--- #57 display.EntityRender
       |--- #58 display.EntityRender
       |--- #59 display.EntityRender
       |--- #60 display.EntityRender
       |--- #61 display.EntityRender
       |--- #62 display.EntityRender
       |--- #63 display.EntityRender
       |--- #64 display.EntityRender
       |--- #65 display.EntityRender
       |--- #66 display.EntityRender
       |--- #67 display.EntityRender
       |--- #68 display.EntityRender
       |--- #69 display.EntityRender
       |--- #70 display.EntityRender
       |--- #71 display.EntityRender
       |--- #72 display.EntityRender
       |--- #73 display.EntityRender
       |--- #74 display.EntityRender
       |--- #75 display.EntityRender
       |--- #76 display.EntityRender
       |--- #77 display.EntityRender
       |--- #78 display.EntityRender
       |--- #79 display.EntityRender
       |--- #80 display.EntityRender
       |--- #81 display.EntityRender
       |--- #82 display.EntityRender
       |--- #83 display.EntityRender
       |--- #90 ui.modal.panel.EditEntityDefs
 |--- #93 page.CrashReport

[FILE]         Cleared all file watches
```

</p>
</details>

**Comment by mazett96** [comment]
Also, now it keeps happening without me changing anything...

**Comment by deepnight** [comment]
Sorry for this one, I'm now pushing a fix in 1.4.1
